artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQMessage.java

```diff
@@ -371,7 +369,7 @@ public Destination getJMSReplyTo() throws JMSException {
             String name = address.toString();
 
             // swap the old prefixes for the new ones so the proper destination type gets created
-            if (enable1xPrefixes) {
+            if (session.isEnable1xPrefixes()) {
```

Comment:
Hot path here

Reply:
It's the same.. instead of a boolean per message, I'm using a single property on the session now. 

We can improve it later to make sure it's static, but I can't do it ATM becuase of RA and caches of connections.

Reply:
The more invasive all these changes just for support of 1.x (isnt that the point of the major version change to signal breaking changes)

Would it maybe better to have a 1x jms client module, that has all these bits in so users on 2.0.x dont get affected and also then keeps it all a bit more clean room